### PR TITLE
ENH: Socket control drr visibility toggle

### LINKED ISSUE #247: DRR visibility - major issues in batch (socket) processing
Unfortunately, the DRR checkbox function is a problem.
Even though the check box is selected - the drr is not 'enabled' at cfg launch (still appears hidden in image views). This means that the optimization merely copies the pose to each frame without tracking. 

Can we:
- [x]  make all volumes check *and* enabled by default?
 and/or
- [x]  allow socket-based enable functionality?
- [x]  remove the null buffer fill in RayCaster if not visible

//if (!visible_) {
    //    cudaMemset(buffer, 0, width * height * sizeof(float));
    //    return;
    //}

### PR DESCRIPTION
**Pull request by NicerNewerCar**
- Closes #247 
- Adds Socket control for toggling the visibility of DRRs
- Defaults the visibility flag in the CUDA backend to `true`
- @amymmorton I don't have an active MATLAB license anymore, could you test the updated class?
   -  ~~**Note:** I haven't set the UI to update yet so the checkbox will still be checked~~

ToDo:
- [x] Get UI to update from the socket 
- [ ] Figure out if #246 is related to the checkbox

### CODE REVIEW

File: libautoscoper/src/gpu/cuda/RayCaster.cpp [hunk 1]
```diff
@@ -61,6 +61,7 @@ RayCaster::RayCaster() : volumeDescription_(0),
     std::stringstream name_stream;
     name_stream << "DrrRenderer" << (++num_ray_casters);
     name_ = name_stream.str();
+    visible_ = true;
```
Comment (jcfr) [inline libautoscoper/src/gpu/cuda/RayCaster.cpp:64]: Consider moving this commit its own PR and reference the commit it is following up on.

Comment (NicerNewerCar) [reply libautoscoper/src/gpu/cuda/RayCaster.cpp]: See #251

**Review by jcfr — APPROVED** [review]

**Review by jcfr — APPROVED** [review]
